internal/pkg/config/config.go

```diff
@@ -64,15 +65,16 @@ func (c *Config) ReplaceInjectionConfigs(replacementConfigs []InjectionConfig) {
 func (c *Config) HasInjectionConfig(key string) bool {
 	c.RLock()
 	defer c.RUnlock()
-	_, ok := c.Injections[key]
+	_, ok := c.Injections[strings.ToLower(key)]
 	return ok
 }
 
 // GetInjectionConfig returns the InjectionConfig given a requested key
 func (c *Config) GetInjectionConfig(key string) (*InjectionConfig, error) {
 	c.RLock()
 	defer c.RUnlock()
-	i, ok := c.Injections[key]
+	k := strings.ToLower(key)
+	i, ok := c.Injections[k]
```

Comment:
note that `i` here would be a copy of `c.Injections[k]`, not a reference, so if `c.Injections[k].(Name|Containers|Volumes|Environment|VolumeMounts)` ever change value, `i` will not get updated, and the returned pointer from this function would essentially be stale. Either the docs need to reflect this or the functionality needs to be adjusted.

Reply:
Very good catch! I am refactoring this to use pointers, so this should be ameliorated shortly. See latest commit